sdks/python/src/opik/api_objects/span.py

```diff
@@ -92,9 +92,13 @@ def update(
         Returns:
             None
         """
-        usage = validation_helpers.extract_supported_usage_data_and_print_result(
-            usage, LOGGER
-        )
+        parsed_usage = validation_helpers.validate_and_parse_usage(usage, LOGGER)
+        if parsed_usage.full_usage is not None:
+            metadata = (
+                {"usage": parsed_usage.full_usage}
+                if metadata is None
+                else {"usage": parsed_usage.full_usage, **metadata}
+            )
```

Comment:
Minor: DRY here, you can encapsulate it in an aux method. There are multiple occurrences of it in this PR.

Reply:
I've created a ticket to tackle later.